app/services/structure_updater.rb

```diff
@@ -100,15 +101,26 @@ def from_csv
     end
 
     contains = files_by_fileset.keys.map do |sequence|
-      # Find the existing fileset
-      existing_fileset = model.structural.contains[sequence.to_i - 1]
+      fileset = fileset_for(sequence, files_by_fileset[sequence].first.label)
       attributes = fileset_attributes[sequence]
                    .merge(structural: { contains: files_by_fileset[sequence] })
-      existing_fileset.new(**attributes)
+      fileset.new(**attributes)
     end
     Success(model.structural.new(contains:))
   end
 
+  FILESET_NAMESPACE = 'https://cocina.sul.stanford.edu/fileset/'
+
+  # @param [int] sequence is the index of the fileset in the import
+  # @param [string] label the label to inject for a new fileset
+  def fileset_for(sequence, label)
+    model.structural.contains[sequence.to_i - 1].presence ||
+      Cocina::Models::FileSet.new(externalIdentifier: "#{FILESET_NAMESPACE}#{druid}-#{SecureRandom.uuid}",
```

Comment:
I don't think there is any reason to include the druid here.

Reply:
No? I thought our pattern was `../driud-uuid` but if not, happy to remove.

Reply:
Cool. That would negate your above comment then, cleaning up, will push shortly.